dev/Tiles/entity.h

```diff
@@ -0,0 +1,66 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include "metatile.h"
+
+typedef struct Entity {
+    MetaTile* metatile[32];
+    unsigned char frameRate;
+    unsigned char numFrames;
+    unsigned int currentFrame;
+} Entity;
+
+Entity* Entity_Create(MetaTile* metatile[], unsigned char numFrames) 
+{
+    Entity* entity = malloc(sizeof(struct Entity));
+    // entity->metatile = malloc(sizeof(struct MetaTile) * 32);
+    entity->currentFrame = 0;
+    entity->frameRate = 30;
+    entity->numFrames = numFrames;
+
+
+    unsigned char i;
+    for(i = 0; i < metatile[i]->numTiles; ++i)
```

Comment:
Two things here:

1. You definitely shouldn't be using a `char` here. As alluded to above somewhere, for loops should almost always use `int` as the index variable, unless you are genuinely looping over some hardware-constrained region, which you aren't here (`metatile` is in fact 32 entries long currently so it doesn't matter at all what integer type you use to iterate over it). Also using `unsigned` ints for index variables is notoriously problematic because comparison with zero usually do unintuitive things.
2. Don't declare your variable outside the loop. In the "old days" (i.e. K&R and C90) you could only declare variables immediately after the opening brace of a block (ln 14 for you here) but from C99 onwards you can - and probably should - declare variables anywhere you like. In this case why should you move the declaration of `i` inside the loop? Because you only use `i` inside the loop and so it's best to reduce the scope of `i` to its minimum to prevent future accidental bugs. On rare occasions you might want to access the final value of an index variable outside of the loop, but more likely it would be a bug.

```suggestion
    for(int i = 0; i < metatile[i]->numTiles; ++i)
```

Reply:
Noted ✅️